src/rpc/blockchain.cpp

```diff
@@ -2240,17 +2241,57 @@ class BlockFiltersScanReserver
     }
 };
 
+static bool CheckBlockFilterMatches(BlockManager& blockman, const CBlockIndex* pindex, const GCSFilter::ElementSet& needles)
+{
+
+    CBlock block;
+    CBlockUndo block_undo;
+
+    {
+        LOCK(cs_main);
+        block = GetBlockChecked(blockman, pindex);
+        if (pindex->nHeight > 0) {
+            block_undo = GetUndoChecked(blockman, pindex);
+        }
```

Comment:
After rebasing on master, excluding the genesis block is not needed anymore, as `GetUndoChecked` takes care of this now and simply returns an empty `CBlockUndo` instance (since #19888, commit 2ca5a496c2f3cbcc63ea15fa05c1658e7f527bbc has been merged):
```suggestion
        block_undo = GetUndoChecked(blockman, pindex);
```